lib/iris/cube.py

```diff
@@ -2302,7 +2306,8 @@ def _as_list_of_coords(self, names_or_coords):
         Convert a name, coord, or list of names/coords to a list of coords.
         """
         # If not iterable, convert to list of a single item
-        if not hasattr(names_or_coords, '__iter__'):
+        if (not hasattr(names_or_coords, '__iter__') or
+                isinstance(names_or_coords, str)):
```

Comment:
I don't believe so - this is just checking whether the input is of type `string` or `list`. The check on whether `names_or_coords` contains one or more `Coord`s happens [on lines 2315-2316](https://github.com/QuLogic/iris/blob/py3k-unicode/lib/iris/cube.py#L2315-L2316).